lib/posix-process/signal/sigaltstack.c

```diff
@@ -0,0 +1,58 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <stddef.h>
+#include <string.h>
+#include <sys/types.h>
+
+#include <uk/process.h>
+#include <uk/syscall.h>
+
+#include "signal.h"
+
+UK_SYSCALL_R_DEFINE(int, sigaltstack, const stack_t *, ss,
+		    stack_t *, old_ss)
+{
+	struct posix_process *proc;
+
+	proc = uk_pprocess_current();
+	UK_ASSERT(proc);
+
+	if (unlikely(ss && ss->ss_flags &&
+		     !((unsigned int)ss->ss_flags == SS_AUTODISARM ||
+		       (unsigned int)ss->ss_flags == SS_DISABLE)))
+		return -EINVAL;
+
+	if (ss && (unsigned int)ss->ss_flags == SS_DISABLE) {
+		proc->signal->altstack.ss_flags |= SS_DISABLE;
+		return 0;
+	}
+
+	if (ss && (unsigned int)ss->ss_flags == SS_AUTODISARM)
+		uk_pr_warn("SS_AUTODISARM stubbed\n");
+
+	if (ss && ss->ss_size < MINSIGSTKSZ)
```

Comment:
Should these error checks use `unlikely`? Also I see that all checks here check for non-NULL `ss` every time. Maybe we can put them all under one check for `ss` that contains all the other checks? E.g
```
if (ss) {
     if (unlikely(..)
     if (unlikely(..)
}
```
I also see one check at the end. Maybe it's just me but just now at a closer look it looks like there may be too many `ss` checks that we could avoid or restructure.

Reply:
Normally I would have argued that this very subjective, as in general I find big conditionals particularly ugly, however on a second look I'm not sure how the compiler will optimize cases with the optional `ss` parameter being inside `unlikely()` the way we would expect it to, so just to be on the safe side, I will move it all into a big ugly `if()` block.